compile error on Mac OS X use pip(python 3.6)

Running setup.py install for librabbitmq ... error
    Complete output from command /usr/local/opt/python3/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/private/tmp/pip-okezcxcl-build/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-l_9g0rqb-record/install-record.txt --single-version-externally-managed --compile:
    running build
    - configure rabbitmq-c...
    /bin/sh: configure: No such file or directory
    running install
    running install_lib
    running build_py
    creating build
    creating build/lib.macosx-10.12-x86_64-3.6
    creating build/lib.macosx-10.12-x86_64-3.6/librabbitmq
    copying librabbitmq/__init__.py -> build/lib.macosx-10.12-x86_64-3.6/librabbitmq
    creating build/lib.macosx-10.12-x86_64-3.6/funtests
    copying funtests/config.py -> build/lib.macosx-10.12-x86_64-3.6/funtests
    copying funtests/disabled_basic_get_leak.py -> build/lib.macosx-10.12-x86_64-3.6/funtests
    copying funtests/__init__.py -> build/lib.macosx-10.12-x86_64-3.6/funtests
    copying funtests/setup.py -> build/lib.macosx-10.12-x86_64-3.6/funtests
    creating build/lib.macosx-10.12-x86_64-3.6/librabbitmq/tests
    copying librabbitmq/tests/test_functional.py -> build/lib.macosx-10.12-x86_64-3.6/librabbitmq/tests
    copying librabbitmq/tests/__init__.py -> build/lib.macosx-10.12-x86_64-3.6/librabbitmq/tests
    creating build/lib.macosx-10.12-x86_64-3.6/funtests/tests
    copying funtests/tests/__init__.py -> build/lib.macosx-10.12-x86_64-3.6/funtests/tests
    copying funtests/tests/test_channel_errors.py -> build/lib.macosx-10.12-x86_64-3.6/funtests/tests
    running build_ext
    building '_librabbitmq' extension
    creating build/temp.macosx-10.12-x86_64-3.6
    creating build/temp.macosx-10.12-x86_64-3.6/Modules
    creating build/temp.macosx-10.12-x86_64-3.6/Modules/_librabbitmq
    creating build/temp.macosx-10.12-x86_64-3.6/rabbitmq-c
    creating build/temp.macosx-10.12-x86_64-3.6/rabbitmq-c/librabbitmq
    clang -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Irabbitmq-c/librabbitmq -Irabbitmq-c -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/include/python3.6m -c Modules/_librabbitmq/connection.c -o build/temp.macosx-10.12-x86_64-3.6/Modules/_librabbitmq/connection.o
    Modules/_librabbitmq/connection.c:299:29: warning: comparison of integers of different signs: 'uint64_t' (aka 'unsigned long long') and 'int' [-Wsign-compare]
                if (clong_value == -1)
                    ~~~~~~~~~~~ ^  ~~
    Modules/_librabbitmq/connection.c:365:17: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
        while (item = PyIter_Next(iterator)) {
               ~~~~~^~~~~~~~~~~~~~~~~~~~~~~
    Modules/_librabbitmq/connection.c:365:17: note: place parentheses around the assignment to silence this warning
        while (item = PyIter_Next(iterator)) {
                    ^
               (                           )
    Modules/_librabbitmq/connection.c:365:17: note: use '==' to turn this assignment into an equality comparison
        while (item = PyIter_Next(iterator)) {
                    ^
                    ==
    Modules/_librabbitmq/connection.c:410:1: warning: unused label 'error' [-Wunused-label]
    error:
    ^~~~~~
    Modules/_librabbitmq/connection.c:1003:16: warning: incompatible pointer to integer conversion returning 'PyObject *' (aka 'struct _object *') from a function with result type 'int' [-Wint-conversion]
            return PyErr_NoMemory();
                   ^~~~~~~~~~~~~~~~
    Modules/_librabbitmq/connection.c:1359:13: warning: implicit declaration of function 'amqp_simple_wait_frame_on_channel' is invalid in C99 [-Wimplicit-function-declaration]
                amqp_simple_wait_frame_on_channel(conn, cur_channel, &frame);
                ^
    Modules/_librabbitmq/connection.c:1359:13: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    6 warnings generated.
    clang -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Irabbitmq-c/librabbitmq -Irabbitmq-c -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/include/python3.6m -c rabbitmq-c/librabbitmq/amqp_api.c -o build/temp.macosx-10.12-x86_64-3.6/rabbitmq-c/librabbitmq/amqp_api.o
    In file included from rabbitmq-c/librabbitmq/amqp_api.c:48:
    rabbitmq-c/librabbitmq/amqp_private.h:332:1: error: expected ')'
    DECLARE_XTOXLL(hton)
    ^
    rabbitmq-c/librabbitmq/amqp_private.h:298:26: note: expanded from macro 'DECLARE_XTOXLL'
      static inline uint64_t func##ll(uint64_t val)     \
                             ^
    <scratch space>:400:1: note: expanded from here
    htonll
    ^
    /usr/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
    #define htonll(x)       __DARWIN_OSSwapInt64(x)
                            ^
    /usr/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
        (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                                 ^
    rabbitmq-c/librabbitmq/amqp_private.h:332:1: note: to match this '('
    rabbitmq-c/librabbitmq/amqp_private.h:298:26: note: expanded from macro 'DECLARE_XTOXLL'
      static inline uint64_t func##ll(uint64_t val)     \
                             ^
    <scratch space>:400:1: note: expanded from here
    htonll
    ^
    /usr/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
    #define htonll(x)       __DARWIN_OSSwapInt64(x)
                            ^
    /usr/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
        (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
        ^
    In file included from rabbitmq-c/librabbitmq/amqp_api.c:48:
    rabbitmq-c/librabbitmq/amqp_private.h:332:1: error: definition of builtin function '__builtin_constant_p'
    DECLARE_XTOXLL(hton)
    ^
    rabbitmq-c/librabbitmq/amqp_private.h:298:26: note: expanded from macro 'DECLARE_XTOXLL'
      static inline uint64_t func##ll(uint64_t val)     \
                             ^
    <scratch space>:400:1: note: expanded from here
    htonll
    ^
    /usr/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
    #define htonll(x)       __DARWIN_OSSwapInt64(x)
                            ^
    /usr/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
        (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
         ^
    In file included from rabbitmq-c/librabbitmq/amqp_api.c:48:
    rabbitmq-c/librabbitmq/amqp_private.h:333:1: error: expected ')'
    DECLARE_XTOXLL(ntoh)
    ^
    rabbitmq-c/librabbitmq/amqp_private.h:298:26: note: expanded from macro 'DECLARE_XTOXLL'
      static inline uint64_t func##ll(uint64_t val)     \
                             ^
    <scratch space>:400:1: note: expanded from here
    ntohll
    ^
    /usr/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
    #define ntohll(x)       __DARWIN_OSSwapInt64(x)
                            ^
    /usr/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
        (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                                 ^
    rabbitmq-c/librabbitmq/amqp_private.h:333:1: note: to match this '('
    rabbitmq-c/librabbitmq/amqp_private.h:298:26: note: expanded from macro 'DECLARE_XTOXLL'
      static inline uint64_t func##ll(uint64_t val)     \
                             ^
    <scratch space>:400:1: note: expanded from here
    ntohll
    ^
    /usr/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
    #define ntohll(x)       __DARWIN_OSSwapInt64(x)
                            ^
    /usr/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
        (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
        ^
    In file included from rabbitmq-c/librabbitmq/amqp_api.c:48:
    rabbitmq-c/librabbitmq/amqp_private.h:333:1: error: definition of builtin function '__builtin_constant_p'
    DECLARE_XTOXLL(ntoh)
    ^
    rabbitmq-c/librabbitmq/amqp_private.h:298:26: note: expanded from macro 'DECLARE_XTOXLL'
      static inline uint64_t func##ll(uint64_t val)     \
                             ^
    <scratch space>:400:1: note: expanded from here
    ntohll
    ^
    /usr/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
    #define ntohll(x)       __DARWIN_OSSwapInt64(x)
                            ^
    /usr/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
        (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
         ^
    4 errors generated.
    error: command 'clang' failed with exit status 1



I use command "sudo pip3 install git+git://github.com/celery/librabbitmq.git@2.0.0", the above error happened, but when I download the source code and execute "sudo make install" command, their is no any error happened, I don't know why？

--- a/setup.py
+++ b/setup.py
@@ -5,9 +5,11 @@ from setuptools import setup, find_packages
 
 # --with-librabbitmq=<dir>: path to librabbitmq package if needed
 
-LRMQDIST = lambda *x: os.path.join('rabbitmq-c', *x)
+BASE_PATH = os.path.dirname(__file__)
+
+LRMQDIST = lambda *x: os.path.join(BASE_PATH, 'rabbitmq-c', *x)
 LRMQSRC = lambda *x: LRMQDIST('librabbitmq', *x)
-PYCP = lambda *x: os.path.join('Modules', '_librabbitmq', *x)
+PYCP = lambda *x: os.path.join(BASE_PATH, 'Modules', '_librabbitmq', *x)
 
 
 def senv(*k__v, **kwargs):
@@ -162,7 +164,7 @@ def find_make(alt=('gmake', 'gnumake', 'make', 'nmake')):
                 return make
 
 
-long_description = open('README.rst', 'U').read()
+long_description = open(os.path.join(BASE_PATH, 'README.rst'), 'U').read()
 distmeta = open(PYCP('distmeta.h')).read().strip().splitlines()
 distmeta = [item.split('\"')[1] for item in distmeta]
 version = distmeta[0].strip()

